lib/DxcSupport/dxcmem.cpp

```diff
@@ -98,3 +98,39 @@ DxcThreadMalloc::DxcThreadMalloc(IMalloc *pMallocOrNull) throw() {
 DxcThreadMalloc::~DxcThreadMalloc() {
     DxcSwapThreadMalloc(pPrior, nullptr);
 }
+
+void* DxcNew(std::size_t size) throw() {
+  void *ptr;
+  IMalloc* iMalloc = DxcGetThreadMallocNoRef();
+  if (iMalloc != nullptr) {
+    ptr = iMalloc->Alloc(size);
+  } else {
+    // DxcGetThreadMallocNoRef() returning null means the operator is called before DllMain
+    // where the g_pDefaultMalloc is initialized, for example from CRT libraries when
+    // static linking is enabled. In that case fallback to the standard allocator
+    // and use CoTaskMemAlloc directly instead of CoGetMalloc, Alloc & Release for better perf.
+#if defined(_WIN32) && !defined(DXC_DISABLE_ALLOCATOR_OVERRIDES)
```

Comment:
Not strictly necessary for non-win platforms as they already have `CoTaskMemAlloc` and `CoTaskMemFree` defined to malloc and free in WinAdapter.h:
https://github.com/microsoft/DirectXShaderCompiler/blob/93555aa911171097d72036df15b2e3c5c0c3d893/include/dxc/WinAdapter.h#L46

I'm unsure if this is needed for `DXC_DISABLE_ALLOCATOR_OVERRIDES`. #5296 didn't change the usages of `CoTaskMem*` in `dxcompilerobj.cpp`: 
https://github.com/microsoft/DirectXShaderCompiler/blob/93555aa911171097d72036df15b2e3c5c0c3d893/tools/clang/tools/dxcompiler/dxcompilerobj.cpp#L1510C40-L1510C40